core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -376,6 +376,11 @@ private void handleResponse(RetryConfig<HttpResponse> retryConfig, ClientRequest
                 retryConfig.requiresResponseTrailers() ?
                 RequestLogProperty.RESPONSE_END_TIME : RequestLogProperty.RESPONSE_HEADERS;
 
+        // The RequestLog may be not complete when a decorator returns a response directly.
```

Comment:
I discovered that the `RedirectingClient` also relies on the request log. Let me fix that separately.
https://github.com/line/armeria/blob/6a1ea7ac6703c91bf85bcbc2d0afe0c7db5a00e9/core/src/main/java/com/linecorp/armeria/client/RedirectingClient.java#L197